Ubuntu 18.04 build error since commit #455

### Problem

I have successfully build AliceVision on two machines both running Ubuntu 18.04, 
but since commit #455 I can no longer build AliceVision on both of them.

**Expected behavior:** Build Success

**Actual behavior:** Build Failed

### Steps to Reproduce

1. git clone https://github.com/alicevision/AliceVision.git --recursive
2. mkdir build; cd build
3. cmake -DCMAKE_BUILD_TYPE=Release . ../AliceVision
4. make 

### Versions

* AliceVision branch/version: Develop
* OS: Ubuntu 18.04
* C++ compiler: GCC 6.4.0
### Error Message
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::verify_options(unsigned int, boost::regex_constants::_match_flags)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::perl_matcher<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::construct_init(boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > > const&, boost::regex_constants::_match_flags)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::get_mem_block()'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::put_mem_block(void*)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > >::maybe_assign(boost::match_results<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<boost::sub_match<__gnu_cxx::__normal_iterator<char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > > > const&)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::cpp_regex_traits_implementation<char>::transform_primary[abi:cxx11](char const*, char const*) const'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::get_default_error_string(boost::regex_constants::error_type)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::basic_regex<char, boost::regex_traits<char, boost::cpp_regex_traits<char> > >::do_assign(char const*, char const*, unsigned int)'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::cpp_regex_traits_implementation<char>::transform[abi:cxx11](char const*, char const*) const'
/home/joran/AliceVision/build/Linux-x86_64/libaliceVision_sfmDataIO.so.0: undefined reference to `boost::re_detail_106501::raise_runtime_error(std::runtime_error const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [Linux-x86_64/aliceVision_utils_keyframeSelection-1.0] Error 1
make[1]: *** [src/software/utils/CMakeFiles/aliceVision_utils_keyframeSelection.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2

--- a/src/aliceVision/sfm/CMakeLists.txt
+++ b/src/aliceVision/sfm/CMakeLists.txt
@@ -73,8 +73,6 @@ alicevision_add_library(aliceVision_sfm
     aliceVision_sfmDataIO
     ${Boost_FILESYSTEM_LIBRARY}
     ${CERES_LIBRARIES}
-  PRIVATE_LINKS
-    ${Boost_REGEX_LIBRARY}
 )
 
 # Unit tests

--- a/src/aliceVision/sfmDataIO/CMakeLists.txt
+++ b/src/aliceVision/sfmDataIO/CMakeLists.txt
@@ -34,6 +34,8 @@ alicevision_add_library(aliceVision_sfmDataIO
   PUBLIC_LINKS
     aliceVision_sfmData
     ${Boost_FILESYSTEM_LIBRARY}
+  PRIVATE_LINKS
+    ${Boost_REGEX_LIBRARY}
 )
 
 

